runtime/lua/vim/lsp/_inlay_hint.lua

```diff
@@ -149,26 +149,78 @@ end
 
 --- Clear inlay hints
 ---
----@param client_id integer|nil filter by client_id. All clients if nil
----@param bufnr integer|nil filter by buffer. All buffers if nil
+---@param bufnr (integer|nil) Buffer handle, or nil for current
 ---@private
-function M.clear(client_id, bufnr)
-  local buffers = bufnr and { resolve_bufnr(bufnr) } or vim.tbl_keys(hint_cache_by_buf)
-  for _, iter_bufnr in ipairs(buffers) do
-    M.__explicit_buffers[iter_bufnr] = false
-    local bufstate = hint_cache_by_buf[iter_bufnr]
-    local client_lens = (bufstate or {}).client_hint or {}
-    local client_ids = client_id and { client_id } or vim.tbl_keys(client_lens)
-    for _, iter_client_id in ipairs(client_ids) do
-      if bufstate then
-        bufstate.client_hint[iter_client_id] = {}
-      end
+local function clear(bufnr)
+  bufnr = resolve_bufnr(bufnr or 0)
+  local bufstate = hint_cache_by_buf[bufnr]
+  local client_lens = (bufstate or {}).client_hint or {}
+  local client_ids = vim.tbl_keys(client_lens)
+  for _, iter_client_id in ipairs(client_ids) do
+    if bufstate then
+      bufstate.client_hint[iter_client_id] = {}
     end
-    api.nvim_buf_clear_namespace(iter_bufnr, namespace, 0, -1)
   end
+  api.nvim_buf_clear_namespace(bufnr, namespace, 0, -1)
   vim.cmd('redraw!')
 end
 
+---@private
+local function reset_timer(reset_bufnr)
+  local timer = M.__buffers[reset_bufnr].timer
+  if timer then
+    M.__buffers[reset_bufnr].timer = nil
+    if not timer:is_closing() then
+      timer:stop()
+      timer:close()
+    end
+  end
+end
+
+---@private
+local function make_request(request_bufnr)
+  reset_timer(request_bufnr)
+  M.refresh({ bufnr = request_bufnr })
+end
+
+---@private
+function M.enable(bufnr)
+  bufnr = resolve_bufnr(bufnr or 0)
+  if not (M.__buffers[bufnr] and M.__buffers[bufnr].enabled) then
+    M.__buffers[bufnr] = { enabled = true, timer = nil }
+    M.refresh({ bufnr = bufnr })
+    api.nvim_buf_attach(bufnr, true, {
```

Comment:
You'll also want handlers for `on_reload` and `on_detach`. `on_reload` should clear the cache and namespace, and refire a request. `on_detach` should clear out any cache related to that buffer or there will be a memory leak as more buffers accumulate state.

You may also want an `LspDetach` autocmd that does the same thing (in case LSP is detached from a buffer manually).

Reply:
done